x-pack/plugins/security_solution/public/detections/containers/detection_engine/rules/use_pre_packaged_rules_status.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useCallback } from 'react';
+import { useQuery, useQueryClient } from 'react-query';
+import { useAppToasts } from '../../../../common/hooks/use_app_toasts';
+import { getPrePackagedRulesStatus } from './api';
+import * as i18n from './translations';
+
+const ONE_MINUTE = 60000;
+
+export interface PrePackagedRulesStatusResponse {
+  rulesCustomInstalled: number;
+  rulesInstalled: number;
+  rulesNotInstalled: number;
+  rulesNotUpdated: number;
+  timelinesInstalled: number;
+  timelinesNotInstalled: number;
+  timelinesNotUpdated: number;
+}
+
+export const PRE_PACKAGED_RULES_STATUS_QUERY_KEY = 'prePackagedRulesStatus';
+
+export const usePrePackagedRulesStatus = () => {
+  const { addError } = useAppToasts();
+
+  return useQuery<PrePackagedRulesStatusResponse>(
+    [PRE_PACKAGED_RULES_STATUS_QUERY_KEY],
+    async ({ signal }) => {
+      const response = await getPrePackagedRulesStatus({ signal });
+
+      return {
+        rulesCustomInstalled: response.rules_custom_installed,
+        rulesInstalled: response.rules_installed,
+        rulesNotInstalled: response.rules_not_installed,
+        rulesNotUpdated: response.rules_not_updated,
+        timelinesInstalled: response.timelines_installed,
+        timelinesNotInstalled: response.timelines_not_installed,
+        timelinesNotUpdated: response.timelines_not_updated,
+      };
+    },
+    {
+      staleTime: ONE_MINUTE * 15,
+      onError: (err) => {
+        addError(err, { title: i18n.RULE_AND_TIMELINE_FETCH_FAILURE });
+      },
+    }
+  );
+};
+
+/**
+ * We should use this hook to invalidate the prepackaged rules cache. For
+ * example, rule mutations that affect rule set size, like creation or deletion,
+ * should lead to cache invalidation.
+ *
+ * @returns A rules cache invalidation callback
+ */
+export const useInvalidatePrePackagedRulesStatus = () => {
```

Comment:
Appreciate the comment/context here. This is one of those things the team just needs to know about when working with mutations as it would be easy to add a mutation operation and not know you have to invalidate the PrePackagedStatus. If we guaranteed all rule access was going through a specific hook/API, we could invalidate at that layer, but not much else can be done, so thanks for the context here 👍